### PR TITLE
Update casting-and-type-conversions.md

### PR DESCRIPTION
**Pull request by ghost**
> from an integer to any arbitrary string

replaced by
> from any arbitrary string to an integer
---
> 8 byte integer

and
> 4 bytes on a 32-bit computer

replaced by

> 64-bit integer

and
> 32-bit integer

I know that 4 bytes is equivalent to 32 bits but I don't see the relation with a 32-bit computer.
Furthermore, the int and long types are consistently documented as 32 and 64-bits data types.

and finally
> the compiler implicitly converts the value on the right to a type `long` before assigning it to `bigNum`. 

replaced by
> the compiler implicitly converts the value of `num` on the right to a type `long` before assigning it to `bigNum`.

to be more clear.

### CODE REVIEW

**Review by jzeferino — APPROVED** [review]
LGTM.

File: docs/csharp/programming-guide/types/casting-and-type-conversions.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ helpviewer_keywords:
 ms.assetid: 568df58a-d292-4b55-93ba-601578722878
 ---
 # Casting and Type Conversions (C# Programming Guide)
-Because C# is statically-typed at compile time, after a variable is declared, it cannot be declared again or used to store values of another type unless that type is convertible to the variable's type. For example, there is no conversion from an integer to any arbitrary string. Therefore, after you declare `i` as an integer, you cannot assign the string "Hello" to it, as is shown in the following code.  
+Because C# is statically-typed at compile time, after a variable is declared, it cannot be declared again or used to store values of another type unless that type is convertible to the variable's type. For example, there is no conversion from any arbitrary string to an integer. Therefore, after you declare `i` as an integer, you cannot assign the string "Hello" to it, as is shown in the following code.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/types/casting-and-type-conversions.md]: This change should be reverted. It was correct as written.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for these changes.  

The second is good, using the same units (bits) to compare sizes. The first is not correct, and should be reverted.

Once you make that fix, we'll merge it.

**Comment by johnnyagerard** [comment]
Are we talking about an implicit conversion here?
>For example, there is no conversion from an integer to any arbitrary string.

If it is the case, we should say "there is no implicit conversion". The sentence would be true but still, it would not match the example.

The example
```
int i;  
i = "Hello"; // Error: "Cannot implicitly convert type 'string' to 'int'" 
```
shows that the string "Hello" cannot be implicitly converted to 'int'.

The sentence
>For example, there is no conversion from an integer to any arbitrary string.

is saying the exact opposite.

**Comment by BillWagner** [comment]
@johnnyagerard 

> Are we talking about an implicit conversion here?

This is making the point that C# is statically typed.  For example:

```csharp
var thing = "abc"; // thing is a string
think = 5; // Not valid, thing is a string, not an int
```

As opposed to dynamic languages, such as JavaScript:

```javascript
var thing = "abc"; // thing is a string
thing = 5; // OK, thing is now a number
```

**Comment by johnnyagerard** [comment]
@BillWagner So it was related to the first sentence, alright then.

**Comment by BillWagner** [comment]
Thanks for making the updates @johnnyagerard 

I've approved this, and I'll :shipit: now.

**Review by BillWagner — APPROVED** [review]

**Comment by pkulikov** [comment]
@BillWagner @johnnyagerard you've confused each other with flipped examples :).

In you example, @BillWagner, you initialize a variable of the `string` type and then try to assign an `int` value to it. However, the [current docs text](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions) contains the example with swapped types: first, a variable of type `int` is initialized, and then the `string` value is assigned to it.

So, I think @johnnyagerard was right initially.

Let's see. The  [current documentation](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions) reads:

>Because C# is statically-typed at compile time, after a variable is declared, it cannot be declared again or used to store values of another type unless that type is convertible to the variable's type. For example, there is no conversion from an integer to any arbitrary string. Therefore, after you declare `i` as an integer, you cannot assign the string "Hello" to it, as is shown in the following code.  
>```csharp  
>int i;  
>i = "Hello"; // Error: "Cannot implicitly convert type 'string' to 'int'"  
>```  

The first sentence is generic. Let's read it with the types from the example:
> Because C# is statically-typed at compile time, after a variable is declared **[`int i;`]**, it cannot be declared again or used to store values **["Hello"]** of another type **[`string`]** unless that type **[`string`]** is convertible to the variable's type **[`int`]**.

So, the example doesn't compile because `string` is not implicitly convertable to `int`, and not because "there is not conversion from an integer to any arbitrary string" as the documentation states now.

One more way to show that the current text is not correct. Let's use the `long` type instead of `string` (the example code also doesn't compile if we use `long`). Then the example would read:
> For example, there is no conversion from `int` to `long`. Therefore, after you declare `i` as `int`, you cannot assign the `long` `100L`, to it, as is shown in the following code.  
>```csharp  
>int i;  
>i = 100L; // error CS0266: Cannot implicitly convert type 'long' to 'int'. An explicit conversion exists (are you missing a cast?)
>```  

However, `int` always converts to `long`.

I would rephrase the whole paragraph under consideration in the following way (my changes are **in bold**):
>Because C# is statically-typed at compile time, after a variable is declared, it cannot be declared again or **assigned** a value of another type unless that type is **implicitly** convertible to the variable's type. For example, **the `string` cannot be implicitly converted to `int`**. Therefore, after you declare `i` as an **`int`**, you cannot assign the string "Hello" to it, as **the following code shows:**  
>```csharp  
>int i;  
>i = "Hello"; // Error: "Cannot implicitly convert type 'string' to 'int'"  
>```

**Comment by BillWagner** [comment]
That's a very good analysis @pkulikov Do you want to submit a PR on this? (I already merged this one.)

**Comment by pkulikov** [comment]
@BillWagner please review #7010

**Comment by johnnyagerard** [comment]
@pkulikov Thanks for your review.
